precompiles/erc20/approve.go

```diff
@@ -49,11 +50,9 @@ func (p Precompile) Approve(
 	authorization, expiration, _ := auth.CheckAuthzExists(ctx, p.AuthzKeeper, grantee, granter, SendMsgURL) //#nosec:G703 -- we are handling the error case (authorization == nil) in the switch statement below
 
 	switch {
-	case authorization == nil && amount != nil && amount.Sign() <= 0:
+	case authorization == nil && amount != nil && amount.Sign() < 0:
 		// case 1: no authorization, amount 0 or negative -> error
-		// TODO: (@fedekunze) check if this is correct by comparing behavior with
-		// regular ERC20
-		err = errors.New("cannot approve non-positive values")
+		err = errors.New("cannot approve negative values")
```

Comment:
nit, let's move this error with the others

Reply:
Done 👍